### ISSUE TITLE
false positive for ptr_arg vs. `.clone()`

### ISSUE THREAD

**Issue by llogiq** [rust-lang-nursery/rust-clippy#1157]
Minimum test case:

``` Rust
fn works(a: &Vec<u8>) -> Vec<u8> { a.clone() }
// now following `ptr_arg`s suggestion:
fn fails(a: &[u8]) -> Vec<u8> { a.clone() } // Ouch
```

Solution:
- [ ] avoid linting when the ptr-argument is cloned somewhere (and a `Vec<_>` is required)
- [ ] bonus: detect those instances and suggest additional measures

**Comment by oli-obk** [comment]
We can detect clone and suggest `to_owned`, since that should be absolutely the same.
